source/main.c

```diff
@@ -34,7 +38,7 @@ void scan_all_buttons_held()
     );
 }
 
-void dol_alloc(int size)
+void dol_alloc(u8 **_dol, int size)
```

Comment:
We're a [two-star](https://wiki.c2.com/?ThreeStarProgrammer) codebase now. :)

There is absolutely no reason to use indirection here, simply return the address.